### PR TITLE
OBSDOCS-1151: Add documentation for transform processor

### PR DESCRIPTION
**Pull request by max-cx**
:warning: This PR is for https://github.com/openshift/openshift-docs/tree/dtotel-docs-3.3, not `main`
:warning: No cherry-picks (cherry-picks will be done when the whole branch gets merged into main on the release date).

Version(s):

4.12, 4.13, 4.14, 4.15, 4.16, 4.17

Issue:

https://issues.redhat.com/browse/OBSDOCS-1151

Link to docs preview:

https://78921--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-collector/otel-collector-processors.html#transform-processor_otel-collector-processors

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: observability/otel/otel-collector/otel-collector-processors.adoc
Comment (max-cx) [inline observability/otel/otel-collector/otel-collector-processors.adoc:477]: @rubenvp8510, question: this block is an example of what?

**Review by IshwarKanse — APPROVED** [review]

File: observability/otel/otel-collector/otel-collector-processors.adoc
Comment (rubenvp8510) [reply observability/otel/otel-collector/otel-collector-processors.adoc:477]: Yes, it is a configuration example.

File: observability/otel/otel-collector/otel-collector-processors.adoc
Comment (max-cx) [inline observability/otel/otel-collector/otel-collector-processors.adoc:448]: @rubenvp8510, is this okay?

File: observability/otel/otel-collector/otel-collector-processors.adoc
Comment (max-cx) [inline observability/otel/otel-collector/otel-collector-processors.adoc:469]: @rubenvp8510, is this okay?

**Review by GroceryBoyJr — COMMENTED** [review]
LGTM

**Comment by GroceryBoyJr** [comment]
@max-cx the word "datapoints" is said not to be in the American English dictionary used by vale. I recommend replacing it with two word: "data points". However, this is not part of the changes you have submitted for review, so it's totally your call sir.

File: observability/otel/otel-collector/otel-collector-processors.adoc
Comment (bergerhoffer) [inline observability/otel/otel-collector/otel-collector-processors.adoc:532]: Should be sentence case

File: observability/otel/otel-collector/otel-collector-processors.adoc
Comment (bergerhoffer) [inline observability/otel/otel-collector/otel-collector-processors.adoc:461]: Seems like this could be lowercase, or perhaps it should be `context` type like in line 466?

```suggestion
For each signal type, the processor processes a series of conditions and statements associated with a specific OTTL context type and then executes them in sequence on incoming telemetry data as specified in the configuration.
```

File: observability/otel/otel-collector/otel-collector-processors.adoc
Comment (bergerhoffer) [inline observability/otel/otel-collector/otel-collector-processors.adoc:458]: Nothing to do about it now, but in the future I'd recommend that these each be modules instead of all of the content being directly in the assembly if possible.

For example, you might want to consider alphabetizing this list of processors. But if you did, it would be a lot more copy/paste and look way different when reviewing the PR - as opposed to just changing on single lines of module inclusions to do the reorder.

Again, nothing to change now, just wanted to toss out the suggestion!

File: observability/otel/otel-collector/otel-collector-processors.adoc
Comment (bergerhoffer) [inline observability/otel/otel-collector/otel-collector-processors.adoc:462]: Should "Telemetry" be capitalized or no? I ask because I think in the other parts of the OCP docs it is always capitalized for some reason. For example here: https://docs.openshift.com/container-platform/4.16/support/remote_health_monitoring/opting-out-of-remote-health-reporting.html

Again, not saying that it's right to capitalize, but maybe it's something we should try to sync all places to do one way or the other if possible

**Review by bergerhoffer — COMMENTED** [review]
Added a few comments for consideration in the future, but nothing worthy of holding up the merge over.

Merging!

**Pull request merged**
